File: apps/backend/src/api/manual.py
"""
API endpoints for Manual Presbiteriano 2019.
"""

from fastapi import APIRouter, HTTPException, Query

from src.services.manual_service import get_article, get_manual_structure, search_articles

router = APIRouter(prefix="/manual", tags=["Manual Presbiteriano"])


@router.get("/structure")
async def get_structure():
    """
    Get the complete manual structure (table of contents).
    Returns parts, chapters, and sections without article content.
    """
    return get_manual_structure()


@router.get("/article/{article_id:path}")
async def get_article_by_id(article_id: str):
    """
    Get a specific article by its ID.
    Returns the full article with text, structure, notes, and navigation.
    """
    article = get_article(article_id)
    if not article:
        raise HTTPException(status_code=404, detail="Artigo não encontrado")
    return article


@router.get("/search")
async def search(
    q: str = Query(..., min_length=2, description="Search query"),
    limit: int = Query(20, ge=1, le=100, description="Maximum results"),
):
    """
    Search articles by text content.
    Returns matching articles with excerpts.
    """
    results = search_articles(q, limit)
    return {
        "query": q,
        "count": len(results),
        "results": results,
    }

File: apps/web/src/services/manual.ts
import { api } from '@/lib/api';

export interface ManualMetadata {
    title: string;
    editionYear: number;
    language: string;
}

export interface ManualArticleSummary {
    id: string;
    number: string;
}

export interface ManualSection {
    id: string;
    number: string;
    title: string;
    articles: ManualArticleSummary[];
}

export interface ManualChapter {
    id: string;
    number: string;
    title: string;
    sections: ManualSection[];
    articles: ManualArticleSummary[];
}

export interface ManualPart {
    id: string;
    title: string;
    chapters: ManualChapter[];
}

export interface ManualStructure {
    metadata: ManualMetadata;
    parts: ManualPart[];
    total_articles: number;
}

export interface ArticleNote {
    id: string;
    number: string;
    marker?: string;
    text?: string;
    page?: number;
}

export interface ArticleStructure {
    id: string;
    type: string;
    marker?: string;
    text: string;
    notes?: ArticleNote[];
}

export interface ArticleNavigation {
    previous: { id: string; number: string } | null;
    next: { id: string; number: string } | null;
}

export interface ManualArticle {
    id: string;
    number: string;
    text: string;
    structure: ArticleStructure[];
    notes: ArticleNote[];
    navigation: ArticleNavigation;
}

export interface SearchResult {
    id: string;
    number: string;
    excerpt: string;
    chapter: string;
    section: string | null;
}

export interface SearchResponse {
    query: string;
    count: number;
    results: SearchResult[];
}

export const manualService = {
    getStructure: async (): Promise<ManualStructure> => {
        const response = await api.get('/manual/structure');
        return response.data;
    },

    getArticle: async (articleId: string): Promise<ManualArticle> => {
        const response = await api.get(`/manual/article/${articleId}`);
        return response.data;
    },

    search: async (query: string, limit: number = 20): Promise<SearchResponse> => {
        const response = await api.get('/manual/search', {
            params: { q: query, limit }
        });
        return response.data;
    }
};
